functions/HttpTriggerAPIUsers/__init__.py

```diff
@@ -1,52 +1,136 @@
 import json
-import pypyodbc
+import pyodbc
 import logging
 import os
 import azure.functions as func
 
 def main(req: func.HttpRequest) -> func.HttpResponse:
-    logging.info('Python HTTP Database trigger function processed a request.')
-    
+    logging.info(
+        'Python HTTP trigger for /users function is processing a request.')
+
+    # Check request method
+    method = req.method
+    if not method:
+        logging.critical('No method available')
+        raise Exception('No method passed')
+
+    # Create a new connection
+    logging.debug("Attempting DB connection!")
+    try:
+        conn = get_db_connection()
+    except (pyodbc.DatabaseError, pyodbc.InterfaceError) as e:
+        logging.error("Failed to connect to DB: " + e.args[0])
+        logging.debug("Error: " + e.args[1])
+        return func.HttpResponse(status_code=500)
+        
+    logging.debug("Connection to DB successful!")
+
+    try:
+        # Return results according to the method
+        if method == "GET":
+            logging.debug("Attempting to retrieve users...")
+            all_users_http_response = get_users(conn)
+            logging.debug("Users retrieved successfully!")
+            return all_users_http_response
+
+        elif method == "POST":
+            logging.debug("Attempting to add user...")
+            user_req_body = req.get_json()
+            new_user_id_http_response = add_user(conn, user_req_body)
+            logging.debug("User added successfully!")
+            return new_user_id_http_response
+
+        else:
+            logging.warn(f"Request with method {method} has been recieved, but that is not allowed for this endpoint")
+            return func.HttpResponse(status_code=405)
+
+    #displays erros encountered when API methods were called
+    except Exception as e:
+        return func.HttpResponse("Error: %s" % str(e), status_code=500)
```

Comment:
Would logging.critical be more appropriate here?

Reply:
I think probably the same case as above

Reply:
I am leaning more towards logging.critical